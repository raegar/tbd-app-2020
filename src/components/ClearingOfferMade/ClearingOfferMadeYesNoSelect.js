import React, {useState} from "react";
import RadioButton from "../RadioButton/RadioButton";
import "./ClearingOfferMade2.css";
import { Container, Col, Row } from "react-bootstrap";
import UCPButton from "../Buttons/UCPButton";
import {Redirect} from "react-router-dom";


/*
//var NAME = false;
function GetTestData(selectedValue) {
    if (selectedValue === "No") {
        NAME = true;
    }
};



function GetNextPage(){
  if(NAME)
  {
      return "/ClearingOfferMadeThirdPage";
  }
  else{

     return "/ClearingOfferMadeFourthPage";
};


}*/ 

const offerOptions = [
	{
		label: "Yes", 
		id: "offerYes", 
		value: "Yes" 
	},
	{
		label: "Yes (Partial Requirements met, this will be where student doesn't have all requirements)", 
		id: "offerYesPart", 
		value: "yespart" 
	},
	{
		label: "Yes (Subject to Ell Status)", 
		id:"Status", 
		value: "yesstatus" 
	},
	{ 
		label: "No", 
		id: "offerNo",
		value: "No" 
	},
	{ 
		label: "Interview Required (this may be due to students having industry experience rather than qualifications)", 
		id: "offerInterviewRequired", 
		value: "InterviewRequired" 
	},
	{ 
		label: "RPA (Highlight this with Emma Cave or Zoe", 
		id: "offerInterviewRequiredHighlight", 
		value: "RPARequired" 
	}      
];

function ClearingOfferMadeYesNoSelect() {
	const [state, setState] = useState({
		whenNo: false,
		whenRPA: false,
		whenInterview: false,
	});

	function setWhenNo(whenNo) {
		setState({
			...state,
			whenNo: whenNo
		});
	}
	
	function setWhenRPA(whenRPA) {
		setState({
			...state,
			whenRPA: whenRPA
		});
	}

	function setWhenInterview(whenInterview) {
		setState({
			...state,
			whenInterview: whenInterview
		});
	}
	
	function handleWhenClick(v) {
    	switch (v) {
    		case 'No':
    			setState({
    				whenNo: true,
    				whenRPA: false,
    				whenInterview: false,
    			});
    			break;
                
    		case 'RPARequired':
    			setState({
    				whenNo: true,
    				whenRPA: false,
    				whenInterview: false,
    			});
    			break;

    	    case 'InterviewRequired':
    			setState({
    				whenNo: false,
    				whenRPA: false,
    				whenInterview: true,
    			});
    			break;
                
    		default:
    			setState({
    				whenNo: false,
    				whenRPA: false,
    				whenInterview: false,
    			});
    	}
	}

	function saveSelectedData() {
		//TODO: see if this is actually needed and not just remnants of a mindless copy paste
    	if (document.getElementById("offerYes").checked) {
    		global.ApplicationFormData.offerMade = document.getElementById("offerYes").value;
    	}

    	if (document.getElementById("offerYesPart").checked) {
    		global.ApplicationFormData.offerMade = document.getElementById("offerYesPart").value;
    	}    

    	if (document.getElementById("Status").checked) {
    		global.ApplicationFormData.offerMade = document.getElementById("Status").value;
    	}

    	if (document.getElementById("offerNo").checked) {
    		global.ApplicationFormData.offerMade = document.getElementById("offerNo").value;
    	}

    	if (document.getElementById("offerInterviewRequired").checked) {
    		global.ApplicationFormData.offerMade = document.getElementById("offerInterviewRequired").value;
    	}

    	if (document.getElementById("offerInterviewRequiredHighlight").checked) {
    		global.ApplicationFormData.offerMade = document.getElementById("offerInterviewRequiredHighlight").value;
    	}

    	console.log(global.ApplicationFormData);
	}

	return ( 
		<Container>
			<h1>Offer</h1>
			<h4>* Required</h4>
			<h3>37. Offer made? (YES/NO)</h3>
			<p><i>Only UNCONDITIONAL Offers can be made (unless interview is needed)</i></p>
			<RadioButton
				heading="Yes"
				options={offerOptions}
				name="offer"
				oc={handleWhenClick}
			/>

			{
				(state.whenNo)
					? <Redirect to={{ pathname: "/ClearingOfferMadeThirdPage" }} />
					: <span></span>
			}

			{
				(state.whenRPA)
					? <Redirect to={{ pathname: "/ClearingOfferMadeFourthPage" }} />
					: <span></span>
			}

			{
				(state.whenInterview)
					? <Redirect to={{ pathname: "/ClearingOfferMadeFourthPage" }} />
					: <span></span>
			}
			<Row>
				<Col>
					<UCPButton to='/ClearingOfferMade'
						primary="True"
						className="mediumbutton"
						buttonText="Go Back"
						href="/ClearingOfferMade"
					/>
				</Col>
				
				<Col>
					<div id="confirmButton" onClick={saveSelectedData}>
						<UCPButton to='/ClearingOfferMadeFifthPage'
							primary="True"
							className="mediumbutton"
							buttonText="Confirm and Continue"
						/>
					</div>
				</Col>
			</Row>
		</Container>
	);
}

export default ClearingOfferMadeYesNoSelect;