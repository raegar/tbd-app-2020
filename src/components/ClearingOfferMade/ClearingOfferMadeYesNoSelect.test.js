import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ClearingOfferMadeYesNoSelect from './ClearingOfferMadeYesNoSelect';
import { BrowserRouter as Router} from 'react-router-dom';

//cleans up 
afterEach(cleanup);

it("Renders ClearingOfferMadeYesNoSelect", () => {
    const { asFragment } = render(<Router><ClearingOfferMadeYesNoSelect /></Router>);
    expect(asFragment()).toMatchSnapshot();
});