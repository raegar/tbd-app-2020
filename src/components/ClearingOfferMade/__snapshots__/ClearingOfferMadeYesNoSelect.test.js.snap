// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ClearingOfferMadeSecondPage 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Offer
    </h1>
    <h4>
      * Required
    </h4>
    <h3>
      37. Offer made? (YES/NO)
    </h3>
    <p>
      <i>
        Only UNCONDITIONAL Offers can be made (unless interview is needed)
      </i>
    </p>
    <div
      class="radio-buttons"
    >
      <h4>
        Yes
      </h4>
      <p />
      <div
        class="radio-group"
      >
        <input
          data-testid="offerYes"
          id="offerYes"
          name="offer"
          type="radio"
          value="Yes"
        />
        <label
          for="offerYes"
        >
          Yes
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerYesPart"
          id="offerYesPart"
          name="offer"
          type="radio"
          value="yespart"
        />
        <label
          for="offerYesPart"
        >
          Yes (Partial Requirements met, this will be where student doesn't have all requirements)
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="Status"
          id="Status"
          name="offer"
          type="radio"
          value="yesstatus"
        />
        <label
          for="Status"
        >
          Yes (Subject to Ell Status)
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerNo"
          id="offerNo"
          name="offer"
          type="radio"
          value="No"
        />
        <label
          for="offerNo"
        >
          No
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerInterviewRequired"
          id="offerInterviewRequired"
          name="offer"
          type="radio"
          value="InterviewRequired"
        />
        <label
          for="offerInterviewRequired"
        >
          Interview Required (this may be due to students having industry experience rather than qualifications)
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerInterviewRequiredHighlight"
          id="offerInterviewRequiredHighlight"
          name="offer"
          type="radio"
          value="RPARequired"
        />
        <label
          for="offerInterviewRequiredHighlight"
        >
          RPA (Highlight this with Emma Cave or Zoe
        </label>
      </div>
    </div>
    <span />
    <span />
    <span />
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          href="/ClearingOfferMade"
        >
          <button
            class="mediumbutton"
            data-testid="ucpButton"
            href="/ClearingOfferMade"
            primary="True"
            type="button"
          >
            Go Back​
          </button>
        </a>
      </div>
      <div
        class="col"
      >
        <div
          id="confirmButton"
        >
          <a
            href="/ClearingOfferMadeFifthPage"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              primary="True"
              type="button"
            >
              Confirm and Continue​
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Renders ClearingOfferMadeYesNoSelect 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Offer
    </h1>
    <h4>
      * Required
    </h4>
    <h3>
      37. Offer made? (YES/NO)
    </h3>
    <p>
      <i>
        Only UNCONDITIONAL Offers can be made (unless interview is needed)
      </i>
    </p>
    <div
      class="radio-buttons"
    >
      <h4>
        Yes
      </h4>
      <p />
      <div
        class="radio-group"
      >
        <input
          data-testid="offerYes"
          id="offerYes"
          name="offer"
          type="radio"
          value="Yes"
        />
        <label
          for="offerYes"
        >
          Yes
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerYesPart"
          id="offerYesPart"
          name="offer"
          type="radio"
          value="yespart"
        />
        <label
          for="offerYesPart"
        >
          Yes (Partial Requirements met, this will be where student doesn't have all requirements)
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="Status"
          id="Status"
          name="offer"
          type="radio"
          value="yesstatus"
        />
        <label
          for="Status"
        >
          Yes (Subject to Ell Status)
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerNo"
          id="offerNo"
          name="offer"
          type="radio"
          value="No"
        />
        <label
          for="offerNo"
        >
          No
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerInterviewRequired"
          id="offerInterviewRequired"
          name="offer"
          type="radio"
          value="InterviewRequired"
        />
        <label
          for="offerInterviewRequired"
        >
          Interview Required (this may be due to students having industry experience rather than qualifications)
        </label>
      </div>
      <div
        class="radio-group"
      >
        <input
          data-testid="offerInterviewRequiredHighlight"
          id="offerInterviewRequiredHighlight"
          name="offer"
          type="radio"
          value="RPARequired"
        />
        <label
          for="offerInterviewRequiredHighlight"
        >
          RPA (Highlight this with Emma Cave or Zoe
        </label>
      </div>
    </div>
    <span />
    <span />
    <span />
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          href="/ClearingOfferMade"
        >
          <button
            class="mediumbutton"
            data-testid="ucpButton"
            href="/ClearingOfferMade"
            primary="True"
            type="button"
          >
            Go Back​
          </button>
        </a>
      </div>
      <div
        class="col"
      >
        <div
          id="confirmButton"
        >
          <a
            href="/ClearingOfferMadeFifthPage"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              primary="True"
              type="button"
            >
              Confirm and Continue​
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
